alphagov/verify-proxy-node
# BAU: Add local startup configuration

This adds the ability to easily run the Proxy Node apps locally with minimal set-up using Docker. There are certain limitations to this at the moment, described in the instructions.

Recommend reviewing the `local-startup/running-proxy-node-locally.md` file first.

  - Provide Dockerfiles for local running
  - Provide `docker.env` file to set up apps for local running
  - Add `docker-compose.yml` file to run Proxy Node apps easily
  - Add script to start up Proxy Node
  - Add local running instructions
  - Update gradle to latest version
  - Add metadata files - these have been manually signed with `xmlsectool` and cannot be changed without re-signing
  - Add metadata truststores - these have been generated manually with `keytool` and need to be re-generated if metadata gets re-signed with a different cert chain
  - Add local PKI (keys and certs symlinked from `verify-dev-pki`)
  - Use locally built artefacts in Docker images by default 
  - Pull VSP image from Proxy Node registry rather than building it from scratch

To test a local journey run `./docker-startup.sh --build` and open `http://localhost:6610/` in your browser. See README for instructions on how to deal with the Compliance Tool.
